src/routes/sessions/[id]/+page.svelte

```diff
@@ -202,6 +200,41 @@
 		session.messages = [...session.messages, message];
 	}
 
+	function resetPrompt() {
+		// Reset the prompt to the last sent message
+		prompt = promptCached;
+		promptCached = '';
+	}
+
+	function abortOllama() {
+		ollamaInstance?.abort();
+		completion = '';
+		// Remove the "incomplete" AI response
+		session.messages = session.messages.slice(0, -1);
+	}
```

Comment:
A few things we can improve about the stop response logic:
- Let's rename this function to `stopResponse()` so it's more descriptive of the _user behavior_, rather than the _technical implementation_.
- There is some code repetition with the `isWaitingForResponse` logic, so let's move it inside `stopResponse()`
- Once we move `if (isWaitingForResponse)`, we will also need to pass an optional argument to handle the beforeNavigate logic: `stopResponse(navigation?: BeforeNavigate)` for `navigate.cancel()`
- Because the toast alert is tied to the abort behavior we should move the toast event to this function
  - Also, it should be a `toast.warn()` because it's not an error.
- Should we also reset the prompt if you navigate out? if so, we might want to call `resetPrompt()` directly inside `stopResponse()`

Should look more or less like this:

```typescript
	import type { BeforeNavigate } from '@sveltejs/kit';

	function stopResponse(navigation?: BeforeNavigate) {
		if (isWaitingForResponse) {
			toast.warning("Can't stop completion until the first response from Ollama is received");
			navigation?.cancel();
			return;
		}

		resetPrompt();
		ollamaInstance?.abort();
		completion = '';
		session.messages = session.messages.slice(0, -1); // Remove the "incomplete" AI response
	}
```